apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts

```diff
@@ -67,20 +67,52 @@ export class SendMessageInApp extends SendMessageType {
     const messagePayload = Object.assign({}, command.payload);
     delete messagePayload.attachments;
 
-    const message = await this.messageRepository.create({
+    const oldMessage = await this.messageRepository.findOne({
       _notificationId: notification._id,
       _environmentId: command.environmentId,
       _organizationId: command.organizationId,
       _subscriberId: command.subscriberId,
       _templateId: notification._templateId,
       _messageTemplateId: inAppChannel.template._id,
       channel: ChannelTypeEnum.IN_APP,
-      cta: inAppChannel.template.cta,
       transactionId: command.transactionId,
-      content,
-      payload: messagePayload,
     });
 
+    let message;
+
+    if (!oldMessage) {
+      message = await this.messageRepository.create({
+        _notificationId: notification._id,
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        _subscriberId: command.subscriberId,
+        _templateId: notification._templateId,
+        _messageTemplateId: inAppChannel.template._id,
+        channel: ChannelTypeEnum.IN_APP,
+        cta: inAppChannel.template.cta,
+        transactionId: command.transactionId,
+        content,
+        payload: messagePayload,
+      });
+    }
+
+    if (oldMessage) {
+      await this.messageRepository.update(
+        {
+          _id: oldMessage._id,
+        },
+        {
+          $set: {
+            seen: false,
+            cta: inAppChannel.template.cta,
+            content,
+            payload: messagePayload,
```

Comment:
@davidsoderberg should we also update the date here? So it will update in the feed location to be on top ?